.github/workflows/backend.yml

```diff
@@ -18,9 +18,9 @@
 name: Backend
 on:
   push:
+  pull_request:
```

Comment:
Do we need to remove below setting?
```
branches:
       - dev
```

This will cause only submit PR to dev will trigger the workflow?

Reply:
> Do we need to remove below setting?
> 
> ```
> branches:
>        - dev
> ```
> 
> This will cause only submit PR to dev will trigger the workflow?

@EricJoy2048 @hailin0 @Hisoka-X @ic4y PTAL

Reply:
> > Do we need to remove below setting?
> > ```
> > branches:
> >        - dev
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This will cause only submit PR to dev will trigger the workflow?
> 
> @EricJoy2048 @hailin0 @Hisoka-X @ic4y PTAL

@TyrantLucifer

Reply:
> Do we need to remove below setting?
> 
> ```
> branches:
>        - dev
> ```
> 
> This will cause only submit PR to dev will trigger the workflow?

I think there is no problem with the dev branch, we are mainly developing in the dev branch now.

Reply:
If we cp to release-branch, the ci will not trigger, it would be better to keep the origin rule.
Why we add the new rule "only trigger on dev"?

Reply:
only trigger on dev rule is added on this pr: https://github.com/apache/seatunnel/pull/4522/files.

For this pr, my purpose is enable the CI on fork repo. Then it can run CI on fork repo first to check, fix error to reduce everyone's time